include/envoy/ssl/context.h

```diff
@@ -32,12 +32,15 @@ class Context {
    */
   virtual std::string getCertChainInformation() const PURE;
 };
+typedef std::shared_ptr<Context> ContextSharedPtr;
 
 class ClientContext : public virtual Context {};
 typedef std::unique_ptr<ClientContext> ClientContextPtr;
+typedef std::shared_ptr<ClientContext> ClientContextSharedPtr;
```

Comment:
Do we intended to make all *Context to SharedPtr? What is the usecase of unique_ptrs?

Reply:
unique_ptr can be removed.

Reply:
Actually, I am working on a separate PR of making ssl_socket not to hold context as its member, only use ctx at constructor.   If this works,  we will continue to use unique_ptr for ctx.